packages/@aws-cdk/aws-s3-deployment/test/integ.bucket-deployment.ts

```diff
@@ -78,42 +80,54 @@ class TestBucketDeployment extends cdk.Stack {
       autoDeleteObjects: true, // needed for integration test cleanup
     });
 
-    const noExtractBucketDeployment = new s3deploy.BucketDeployment(this, 'DeployMeWithoutExtractingFilesOnDestination', {
+    new s3deploy.BucketDeployment(this, 'DeployMeWithoutExtractingFilesOnDestination', {
       sources: [s3deploy.Source.asset(path.join(__dirname, 'my-website'))],
       destinationBucket: bucket4,
       extract: false,
       retainOnDelete: false,
     });
 
-    new cdk.CfnOutput(this, 'ObjectKey0', {
-      value: cdk.Fn.select(0, noExtractBucketDeployment.objectKeys),
+    this.bucket5 = new s3.Bucket(this, 'Destination5', {
+      publicReadAccess: false,
+      removalPolicy: cdk.RemovalPolicy.DESTROY,
+      autoDeleteObjects: true, // needed for integration test cleanup
+    });
+
+    const deploy5 = new s3deploy.BucketDeployment(this, 'DeployMe5', {
+      sources: [s3deploy.Source.asset(path.join(__dirname, 'my-website-second'))],
+      destinationBucket: this.bucket5,
+      retainOnDelete: false, // default is true, which will block the integration test cleanup
     });
+    deploy5.addSource(s3deploy.Source.data('some-key', 'helloworld'));
   }
 }
 
 const app = new cdk.App();
 const testCase = new TestBucketDeployment(app, 'test-bucket-deployments-2');
 
 // Assert that DeployMeWithoutExtractingFilesOnDestination deploys a zip file to bucket4
```

Comment:
```suggestion
// Assert that DeployMeWithoutExtractingFilesOnDestination deploys a zip file to bucket5
```

Reply:
This one I'll fix on the main branch just to avoid divergence between the release branch and main.